dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -26,9 +26,9 @@ public abstract class ClientBase
     private const int MaxResultsPerPrompt = 128;
 
     // Prevent external inheritors
-    private protected ClientBase(ILogger? logger = null)
+    private protected ClientBase(ILoggerFactory? loggerFactory = null)
     {
-        this.Logger = logger ?? NullLogger.Instance;
+        this.Logger = loggerFactory is not null ? loggerFactory.CreateLogger(this.GetType().Name) : NullLogger.Instance;
```

Comment:
Curious your thoughts on the following options @stephentoub :

```suggestion
        this.Logger = loggerFactory is not null ? loggerFactory.CreateLogger(this.GetType()) : NullLogger.Instance;
```


```suggestion
        this.Logger = loggerFactory is not null ? loggerFactory.CreateLogger(this.GetType().FullName) : NullLogger.Instance;
```

Reply:
Using `this.GetType()` is a bit more expensive, as CreateLogger needs to format the type into the name. In .NET 8, that's at least reasonably cheap in the most common case; for less common cases, e.g. where the type is generic, it'll involve allocations.

Using `this.GetType().FullName` is a fine choice, and more closely matches what you'd get in the common case with just `this.GetType()`.  I think it ends up coming down to how verbose we want those categories to be. If we want the full name, that's fine, we should just consistently use it in all such places.